x/name/client/cli/gov.go

```diff
@@ -0,0 +1,89 @@
+package cli
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/provenance-io/provenance/x/name/types"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/gov/client/cli"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// The flag for the owner address of a name record
+const flagOwner = "owner"
+
+// GetRootNameProposalCmd returns a command for registration with the gov module
+func GetRootNameProposalCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "root-name-proposal [name] --restrict --owner [address] --title [text] --description [text]",
+		Short: "Submit a root name proposal",
+		Args:  cobra.ExactArgs(1),
```

Comment:
#nit do you want to add an example use case?  Like we do here: [link](https://github.com/provenance-io/provenance/blob/main/x/name/client/cli/tx.go#L43)

Reply:
Added an expanded example